frontend/model/state.js

```diff
@@ -502,6 +489,41 @@ const getters = {
       return identityState && identityState.attributes
     }
   },
+  groupMembers (state, getters) {
+    return contractID => {
+      const profiles = state[contractID]?.profiles || {}
+      return Object.keys(profiles).map(username => {
+        return getters.globalProfile2(contractID, username)
+      })
+    }
+  },
+  ourContactProfiles (state, getters) {
+    const profiles = {}
+    const allProfiles = getters.groupsByName
+      .map(({ groupName, contractID }) => getters.groupMembers(contractID))
+      .flat()
+    const profilesSet = allProfiles
+      .filter((profile, pos) => profile && profile.username !== getters.ourUsername &&
+        allProfiles.findIndex(p => p.username === profile.username) === pos)
```

Comment:
What is this doing and why? I am especially confused by `allProfiles.findIndex(p => p.username === profile.username) === pos`

Also, what is this getter supposed to be returning?

Reply:
`allProfiles.findIndex(p => p.username === profile.username) === pos`

If there is better solution for removing duplications, I will gladly accept it. For example, using Set...

Reply:
@Silver-IT You can loop through `allProfiles` in a single loop and set `profiles[profile.username] = profile` if `profile.username !== getters.ourUsername`.

In other words, you can modify the `for` loop on line 508 below to do that, and then delete the 3 lines above.